src/h5web/visualizations/shared/hooks.ts

```diff
@@ -40,6 +40,18 @@ export function useMappedArray<T>(
   }, [mapperState, baseArray]);
 }
 
+export function useDataArrays<T>(
+  value: T[],
+  rawDims: number[],
+  dimensionMapping: DimensionMapping
+) {
+  const baseArray = useBaseArray(value, rawDims) as ndarray<T>;
+  return {
+    baseArray,
+    dataArray: useMappedArray(baseArray, dimensionMapping) as ndarray<T>,
+  };
+}
```

Comment:
I had to introduce `undefined` in `use...Array` hooks but I cannot make TS infer that the array will be defined if the value in input is defined.

To fix this, I added this hook that makes the necessary assertions when `value` is defined to get arrays that are always defined. I then use this hook in all `Mapped...` component.

Reply:
Right, that will do for now. Might be good to add a comment, though, since it's not very self-documenting. I'm tempted to ask a question on StackOverflow -- it's such a simple problem that I feel there must be a solution.